### PR TITLE
Unity segment improvements

### LINKED ISSUE #3538: Unity C# version not showing (failed web request)
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

The C# version isn't showing in the latest release but works fine when I try in the devcontainer. The debug log shows an error when the web request is made to fetch the Unity documentation webpage.

```
context deadline exceeded
```

 Any ideas?

### Theme

Stripped down version of jandedobbeleer.

### What OS are you seeing the problem on?

Windows

### Which shell are you using?

powershell

### Log output

```Shell
[DEBUG] 21:46:56.156 shell.go:Getenv:377 → C:\Users\colt45\AppData\Local
[TRACE] 21:46:56.156 shell.go:Getenv(LOCALAPPDATA) - 0s
[TRACE] 21:46:56.156 shell_windows.go:CachePath() - 0s
[DEBUG] 21:46:56.158 shell.go:Getenv:377 → C:\github\prompt\jandedobbeleer.omp.json
[TRACE] 21:46:56.158 shell.go:Getenv(POSH_THEME) - 0s
[DEBUG] 21:46:56.165 shell.go:Shell:622 → process name: pwsh.exe
[TRACE] 21:46:56.165 shell.go:Shell() - 7.2241ms
[TRACE] 21:46:56.165 shell.go:resolveConfigPath() - 7.2241ms
[TRACE] 21:46:56.165 shell.go:Init() - 8.8573ms
[TRACE] 21:46:56.165 shell.go:Flags() - 0s
[TRACE] 21:46:56.166 config.go:loadConfig() - 650.3µs
[TRACE] 21:46:56.166 shell.go:Flags() - 0s
[DEBUG] 21:46:56.166 shell.go:Getenv:377 → NO DATA
[TRACE] 21:46:56.166 shell.go:Getenv(OMP_CACHE_DISABLED) - 0s
[TRACE] 21:46:56.166 shell_windows.go:WindowsRegistryKeyValue(HKEY_CURRENT_USER\Software\Microsoft\Windows\DWM\ColorizationColor) - 0s
[TRACE] 21:46:56.166 shell.go:GOOS() - 0s
[TRACE] 21:46:56.166 shell.go:GOOS() - 0s
[TRACE] 21:46:56.166 shell.go:GOOS() - 0s
[TRACE] 21:46:56.166 shell.go:GOOS() - 0s
[TRACE] 21:46:56.168 shell.go:GOOS() - 0s
[TRACE] 21:46:56.168 shell.go:GOOS() - 0s
[TRACE] 21:46:56.168 shell.go:GOOS() - 0s
[TRACE] 21:46:56.168 shell.go:GOOS() - 0s
[TRACE] 21:46:56.168 shell.go:GOOS() - 0s
[TRACE] 21:46:56.168 shell.go:GOOS() - 0s
[TRACE] 21:46:56.168 shell.go:GOOS() - 0s
[TRACE] 21:46:56.168 shell.go:GOOS() - 0s
[TRACE] 21:46:56.168 shell.go:GOOS() - 0s
[TRACE] 21:46:56.168 shell.go:GOOS() - 0s
[TRACE] 21:46:56.168 shell.go:GOOS() - 0s
[TRACE] 21:46:56.168 shell.go:GOOS() - 0s
[TRACE] 21:46:56.168 shell.go:GOOS() - 0s
[TRACE] 21:46:56.168 shell.go:GOOS() - 0s
[TRACE] 21:46:56.168 shell.go:GOOS() - 0s
[DEBUG] 21:46:56.169 shell_windows.go:WindowsRegistryKeyValue:206 → ColorizationColor(DWORD): 0x00632812
[TRACE] 21:46:56.169 shell_windows.go:Root() - 0s
[TRACE] 21:46:56.169 shell.go:Shell() - 0s
[TRACE] 21:46:56.169 shell.go:ErrorCode() - 0s
[TRACE] 21:46:56.169 shell_windows.go:IsWsl() - 0s
[TRACE] 21:46:56.169 shell.go:GOOS() - 0s
[DEBUG] 21:46:56.169 shell.go:Pwd:405 → NO DATA
[TRACE] 21:46:56.169 shell.go:Pwd() - 0s
[DEBUG] 21:46:56.169 shell_windows.go:56 → C:\Users\colt45
[TRACE] 21:46:56.169 shell.go:GOOS() - 0s
[TRACE] 21:46:56.169 shell.go:GOOS() - 0s
[TRACE] 21:46:56.169 shell.go:GOOS() - 0s
[TRACE] 21:46:56.169 shell.go:GOOS() - 0s
[TRACE] 21:46:56.169 shell.go:GOOS() - 0s
[TRACE] 21:46:56.169 shell.go:GOOS() - 0s
[TRACE] 21:46:56.169 shell.go:GOOS() - 0s
[TRACE] 21:46:56.169 shell.go:GOOS() - 0s
[TRACE] 21:46:56.169 shell.go:GOOS() - 0s
[TRACE] 21:46:56.169 shell.go:GOOS() - 0s
[TRACE] 21:46:56.169 shell.go:GOOS() - 0s
[TRACE] 21:46:56.169 shell.go:GOOS() - 0s
[TRACE] 21:46:56.169 shell.go:GOOS() - 0s
[TRACE] 21:46:56.169 shell.go:GOOS() - 0s
[TRACE] 21:46:56.169 shell.go:GOOS() - 0s
[TRACE] 21:46:56.169 shell.go:GOOS() - 0s
[TRACE] 21:46:56.169 shell.go:GOOS() - 0s
[TRACE] 21:46:56.169 shell.go:GOOS() - 0s
[DEBUG] 21:46:56.169 shell.go:User:526 → colt45
[TRACE] 21:46:56.169 shell.go:User() - 0s
[DEBUG] 21:46:56.169 shell.go:Host:538 → COLT-DEV-DELL
[TRACE] 21:46:56.169 shell.go:Host() - 0s
[TRACE] 21:46:56.169 shell.go:GOOS() - 0s
[TRACE] 21:46:56.169 shell.go:TemplateCache() - 0s
[TRACE] 21:46:56.169 shell.go:Flags() - 0s
[DEBUG] 21:46:56.169 shell.go:Pwd:385 → C:\Users\colt45\Desktop\Dev\Unity\Scratch\AndroidSandbox
[TRACE] 21:46:56.169 shell.go:Pwd() - 0s
[DEBUG] 21:46:56.169 shell_windows.go:56 → C:\Users\colt45
[TRACE] 21:46:56.169 shell.go:GOOS() - 0s
[DEBUG] 21:46:56.169 shell.go:Getenv:377 → NO DATA
[TRACE] 21:46:56.169 shell.go:Getenv(SSH_CONNECTION) - 0s
[DEBUG] 21:46:56.169 shell.go:Getenv:377 → NO DATA
[TRACE] 21:46:56.169 shell.go:Getenv(SSH_CLIENT) - 0s
[TRACE] 21:46:56.169 shell.go:TemplateCache() - 0s
[TRACE] 21:46:56.170 shell.go:TemplateCache() - 0s
[TRACE] 21:46:56.170 shell.go:Shell() - 0s
[DEBUG] 21:46:56.170 shell.go:Pwd:385 → C:\Users\colt45\Desktop\Dev\Unity\Scratch\AndroidSandbox
[TRACE] 21:46:56.170 shell.go:Pwd() - 0s
[DEBUG] 21:46:56.170 shell_windows.go:56 → C:\Users\colt45
[TRACE] 21:46:56.170 shell.go:GOOS() - 0s
[DEBUG] 21:46:56.170 shell.go:Pwd:385 → C:\Users\colt45\Desktop\Dev\Unity\Scratch\AndroidSandbox
[TRACE] 21:46:56.170 shell.go:Pwd() - 0s
[TRACE] 21:46:56.170 shell.go:Shell() - 0s
[TRACE] 21:46:56.170 shell.go:Flags() - 0s
[DEBUG] 21:46:56.170 shell_windows.go:56 → C:\Users\colt45
[TRACE] 21:46:56.170 shell.go:GOOS() - 0s
[TRACE] 21:46:56.170 shell.go:GOOS() - 0s
[TRACE] 21:46:56.170 shell.go:PathSeparator() - 0s
[TRACE] 21:46:56.170 shell.go:GOOS() - 0s
[TRACE] 21:46:56.170 shell.go:GOOS() - 0s
[TRACE] 21:46:56.170 shell.go:GOOS() - 0s
[TRACE] 21:46:56.170 shell.go:GOOS() - 0s
[TRACE] 21:46:56.170 shell.go:GOOS() - 0s
[TRACE] 21:46:56.170 shell.go:GOOS() - 0s
[TRACE] 21:46:56.170 shell.go:GOOS() - 0s
[TRACE] 21:46:56.170 shell.go:PathSeparator() - 0s
[TRACE] 21:46:56.170 shell.go:PathSeparator() - 0s
[TRACE] 21:46:56.170 shell.go:GOOS() - 0s
[TRACE] 21:46:56.170 shell.go:GOOS() - 0s
[TRACE] 21:46:56.170 shell.go:GOOS() - 0s
[TRACE] 21:46:56.170 shell.go:PathSeparator() - 0s
[TRACE] 21:46:56.170 shell.go:GOOS() - 0s
[TRACE] 21:46:56.170 shell.go:GOOS() - 0s
[TRACE] 21:46:56.170 shell.go:GOOS() - 0s
[TRACE] 21:46:56.170 shell.go:PathSeparator() - 0s
[TRACE] 21:46:56.170 shell.go:GOOS() - 0s
[TRACE] 21:46:56.170 shell.go:GOOS() - 0s
[TRACE] 21:46:56.170 shell.go:GOOS() - 0s
[TRACE] 21:46:56.170 shell.go:PathSeparator() - 0s
[TRACE] 21:46:56.170 shell.go:PathSeparator() - 0s
[TRACE] 21:46:56.170 shell.go:GOOS() - 0s
[TRACE] 21:46:56.170 shell.go:GOOS() - 0s
[TRACE] 21:46:56.170 shell.go:GOOS() - 0s
[TRACE] 21:46:56.170 shell.go:GOOS() - 0s
[TRACE] 21:46:56.170 shell.go:GOOS() - 0s
[TRACE] 21:46:56.170 shell.go:GOOS() - 0s
[TRACE] 21:46:56.170 shell.go:GOOS() - 0s
[TRACE] 21:46:56.170 shell.go:GOOS() - 0s
[TRACE] 21:46:56.170 shell.go:GOOS() - 0s
[TRACE] 21:46:56.170 shell.go:GOOS() - 0s
[TRACE] 21:46:56.170 shell.go:GOOS() - 0s
[TRACE] 21:46:56.170 shell.go:GOOS() - 0s
[TRACE] 21:46:56.170 shell.go:GOOS() - 0s
[TRACE] 21:46:56.170 shell.go:GOOS() - 0s
[TRACE] 21:46:56.170 shell.go:GOOS() - 0s
[TRACE] 21:46:56.170 shell.go:GOOS() - 0s
[TRACE] 21:46:56.170 shell.go:PathSeparator() - 0s
[TRACE] 21:46:56.170 shell.go:PathSeparator() - 0s
[DEBUG] 21:46:56.170 shell.go:Pwd:385 → C:\Users\colt45\Desktop\Dev\Unity\Scratch\AndroidSandbox
[TRACE] 21:46:56.170 shell.go:Pwd() - 0s
[TRACE] 21:46:56.170 shell_windows.go:IsWsl() - 0s
[TRACE] 21:46:56.170 shell.go:StackCount() - 0s
[DEBUG] 21:46:56.171 win32_windows.go:isWriteable:299 → not current user or in group
[DEBUG] 21:46:56.171 win32_windows.go:isWriteable:303 → current user is member of S-1-5-32-544
[DEBUG] 21:46:56.171 win32_windows.go:isWriteable:311 ↓
    WRITE_DAC
    WRITE_OWNER
    SYNCHRONIZE
    DELETE
    READ_CONTROL
[DEBUG] 21:46:56.171 win32_windows.go:isWriteable:313 → user has write access
[TRACE] 21:46:56.171 shell_windows.go:DirIsWritable() - 621.2µs
[TRACE] 21:46:56.171 shell.go:TemplateCache() - 0s
[TRACE] 21:46:56.171 shell.go:TemplateCache() - 0s
[TRACE] 21:46:56.171 shell.go:Shell() - 0s
[DEBUG] 21:46:56.171 shell.go:Pwd:385 → C:\Users\colt45\Desktop\Dev\Unity\Scratch\AndroidSandbox
[TRACE] 21:46:56.171 shell.go:Pwd() - 0s
[DEBUG] 21:46:56.171 shell_windows.go:56 → C:\Users\colt45
[TRACE] 21:46:56.171 shell.go:GOOS() - 0s
[DEBUG] 21:46:56.171 shell.go:Pwd:385 → C:\Users\colt45\Desktop\Dev\Unity\Scratch\AndroidSandbox
[TRACE] 21:46:56.171 shell.go:Pwd() - 0s
[TRACE] 21:46:56.171 shell.go:HasParentFilePath(ProjectSettings) - 0s
[DEBUG] 21:46:56.171 shell.go:HasFilesInDir:436 → true
[TRACE] 21:46:56.171 shell.go:HasFilesInDir(ProjectVersion.txt) - 503.3µs
[DEBUG] 21:46:56.171 shell.go:FileContent:494 ↓
    m_EditorVersion: 2021.3.16f1
    m_EditorVersionWithRevision: 2021.3.16f1 (4016570cf34f)

[TRACE] 21:46:56.171 shell.go:FileContent(C:\Users\colt45\Desktop\Dev\Unity\Scratch\AndroidSandbox\ProjectSettings\ProjectVersion.txt) - 0s
[DEBUG] 21:46:56.171 shell.go:HTTPRequest:666 ↓
    GET /2021.3/Documentation/Manual/CSharpCompiler.html HTTP/1.1
    Host: docs.unity3d.com
    User-Agent: Go-http-client/1.1
    Accept-Encoding: gzip


[ERROR] 21:46:59.178 shell.go:HTTPRequest:670 → Get "https://docs.unity3d.com/2021.3/Documentation/Manual/CSharpCompiler.html": context deadline exceeded
[TRACE] 21:46:59.178 shell.go:HTTPRequest(https://docs.unity3d.com/2021.3/Documentation/Manual/CSharpCompiler.html) - 3.0067786s
[ERROR] 21:46:59.178 unity.go:Enabled:35 → context deadline exceeded
[TRACE] 21:46:59.178 shell.go:TemplateCache() - 0s
[TRACE] 21:46:59.178 shell.go:TemplateCache() - 0s
[TRACE] 21:46:59.179 shell.go:Shell() - 0s
[DEBUG] 21:46:59.179 shell.go:Getenv:377 → C:\Users\colt45\AppData\Local
[TRACE] 21:46:59.179 shell.go:Getenv(LOCALAPPDATA) - 0s
[TRACE] 21:46:59.179 shell_windows.go:CachePath() - 0s
[TRACE] 21:46:59.179 shell.go:Flags() - 0s
```

### PR DESCRIPTION
**Pull request by cabauman**
### Prerequisites

- [x] I have read and understood the [contributing guide][CONTRIBUTING.md]
- [x] The commit message follows the [conventional commits][cc] guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added/updated (for bug fixes/features)

### Description

Resolves #3538

### CODE REVIEW

File: src/segments/unity.go [hunk 1]
```diff
@@ -19,6 +19,30 @@ type Unity struct {
 	CSharpVersion string
 }
 
+var csharpVersionsByUnityVersion = map[string]string{
```
Comment (JanDeDobbeleer) [inline src/segments/unity.go]: can we add this in `GetCSharpVersion()`? That way the map isn't loaded into memory if you're not using the segment.

File: src/segments/unity.go [hunk 2]
```diff
@@ -76,8 +105,17 @@ func (u *Unity) GetCSharpVersion() (version string, err error) {
...
 	shortUnityVersion := u.UnityVersion[0:lastDotIndex]
 
-	if val, found := u.env.Cache().Get(shortUnityVersion); found {
-		csharpVersion := strings.TrimSuffix(val, ".0")
+	csharpVersion, found := csharpVersionsByUnityVersion[shortUnityVersion]
+	if found {
+		return csharpVersion, nil
+	}
+
+	u.env.Debug(fmt.Sprintf("Unity version %s doesn't exist in the map. Now making web request...", shortUnityVersion))
```
Comment (JanDeDobbeleer) [inline src/segments/unity.go]:
```suggestion
	u.env.Debug(fmt.Sprintf("Unity version %s doesn't exist in the map", shortUnityVersion))
```

**Review by JanDeDobbeleer — CHANGES_REQUESTED** [review]

**Review by JanDeDobbeleer — APPROVED** [review]

**Comment by JanDeDobbeleer** [comment]
@all-contributors please add @cabauman for code,doc

**Comment by cabauman** [comment]
Thanks for your time and help! I love what you've done with this repository.

**Pull request merged**